src/util/usertools.c

```diff
@@ -570,12 +571,32 @@ sss_fqname(char *str, size_t size, struct sss_names_ctx *nctx,
                               name, domain->name, calc_flat_name (domain), NULL);
 }
 
-errno_t sss_user_by_name_or_uid(const char *input, uid_t *_uid, gid_t *_gid)
+errno_t sss_user_by_name_or_uid(struct sss_nss_ops *ops, const char *input,
+                                uid_t *_uid, gid_t *_gid)
 {
     uid_t uid;
     errno_t ret;
     char *endptr;
-    struct passwd *pwd;
+    struct passwd pwd = { 0 };
+    TALLOC_CTX *tmp_ctx;
+    int errnop = 0;
+    enum nss_status status;
+    char s_nss_buffer[NSS_BUFFER_SIZE];
+
+    tmp_ctx = talloc_new(NULL);
```

Comment:
What's the purpose of `tmp_ctx`?

Reply:
It's what remains from a previous implementation. I've removed it.